### PR TITLE
chore(discussions): Always allow users to be owners of discussions

### PR DESCRIPTION
**Pull request by ewinslow**
Will fix https://github.com/Elgg/www.elgg.org/issues/12

### CODE REVIEW

**Comment by ewinslow** [comment]
Tested locally.

**Comment by ewinslow** [comment]
Hmm... maybe this isn't right... Need to check if it is the same as the current logged in user.

**Comment by ewinslow** [comment]
This works, and seems closer to what we're going for.

Wish owner_guid and container_guid permissions checks weren't conflated like this...

File: mod/discussions/start.php [hunk 1]
```diff
@@ -647,7 +647,8 @@ function discussion_topic_container_permissions($hook, $type, $value, $params) {
 	$subtype = elgg_extract('subtype', $params);
 	$container = elgg_extract('container', $params);
 
-	if (!$value || !$container || $subtype !== 'discussion') {
+	if (!$value || !$container || $subtype !== 'discussion' ||
+			$container == elgg_get_logged_in_user_entity()) {
```
Comment (mrclay) [inline mod/discussions/start.php]: This will allow users to create discussions even if the "user" type isn't approved.

**Comment by mrclay** [comment]
Can you briefly explain the container/owner permissions checks?

File: mod/discussions/start.php (hunk 1 above)
Comment (mrclay) [inline mod/discussions/start.php]: Yeah, we don't want this in core.

**Comment by ewinslow** [comment]
More background on this:

https://github.com/Elgg/Elgg/issues/4231

**Comment by ewinslow** [comment]
Basically, when an entity is saved, Elgg checks `can_write_to_container($container)` and `can_write_to_container($owner)`. `can_write_to_container` is responsible for triggering the `container_permissions_check` hook.

**Comment by mrclay** [comment]
Ugh.

**Comment by ewinslow** [comment]
srsly

**Comment by ewinslow** [comment]
I'm looking for a way to make people owners but not containers...

**Comment by mrclay** [comment]
Are you saying master doesn't work?

**Comment by ewinslow** [comment]
Correct, master is broken.

**Comment by mrclay** [comment]
OK I see that now.

**Comment by ewinslow** [comment]
I see a couple ways forward:

 1. Pull this in
 2. Revert #8761 
 3. Figure out how to allow ownership but not containership

**Comment by mrclay** [comment]
Revert #8761. Not the right approach for the perms system as it is today.

**Pull request closed without merging**